pinecone/core/grpc/index_grpc.py

```diff
@@ -259,7 +259,11 @@ def _vector_transform(item):
             return self._wrap_grpc_call(self.stub.Upsert, request, timeout=timeout)
 
     def delete(self, *args, async_req=False, **kwargs):
-        request = DeleteRequest(*args, **kwargs)
+        _filter = dict_to_proto_struct(kwargs.pop('filter', None))
+        filter_param = {}
+        if _filter:
+            filter_param['filter'] = _filter
```

Comment:
@benjaminran  is the param expected to be something like {'filter':'...'} ?

Reply:
It's expected to be a filter= kwarg, like `index.delete(filter={"genre":"documentary"})`. 

I'm using filter_param here to avoid passing the filter= kwarg as struct({}) if the user didn't specify it (so that struct({}) isn't added to the payload). This is because
- it's not adding any value and just adds a bit of size to the payload
- and the backend enforces that filter shouldn't be passed if ids or delete_all==true is; I'd have to add extra logic in the backend to consider struct({}) a value that doesn't count as a filter